atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -614,13 +615,17 @@ private MetricsManager setUpMetricsAndGetMetricsManager() {
     @VisibleForTesting
     @Value.Derived
     String serviceName() {
-        return serviceIdentifierOverride().orElseGet(() -> Stream.of(config().namespace(),
+        return serviceIdentifierOverride().orElseGet(this::namespace);
+    }
+
+    private String namespace() {
```

Comment:
nit: could we make this `Value.Derived`, and/or have the service name take a `Suppliers.memoize(() -> namespace())`? I'm worried we're doing a bunch of unnecessary stream reads every time we look up the namespace.